common/database/connector.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * This singleton class provides a pooled Doctrine connection to consumers. All connection data
+ * is configurable via sqlconf.php.
+ *
+ * If needed, the instance can be used in a transactional context:
+ * <code>
+ *     $database = \common\database\Connector::Instance();
+ *     $entityManager = $database->entityManager;
+ *     $entityManager->getConnection()->beginTransaction();
+ *     try {
+ *         // Entity work here...
+ *         $entityManager->persist($someEntityToBePersisted);
+ *         $entityManager->flush();
+ *         $entityManager->getConnection()->commit();
+ *     } catch (Exception $e) {
+ *         $entityManager->getConnection()->rollBack();
+ *         throw $e;
+ *     }
+ * </code>
+ *
+ * Copyright (C) 2016 Matthew Vita <matthewvita48@gmail.com>
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>.
+ *
+ * @package OpenEMR
+ * @author  Matthew Vita <matthewvita48@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+namespace common\database;
+
+use \Doctrine\ORM\Tools\Setup;
+use \Doctrine\ORM\EntityManager;
+use \common\database\Auditor;
+
+final class Connector {
+    /**
+     * The pooled Doctrine connection.
+     */
+    public $entityManager;
+
+    /**
+     * Logger for noting connection/configuration information.
+     */
+    private $logger;
+
+    /**
+     * Default constructor.
+     */
+    private function __construct() {
+        $this->logger = new \common\logging\Logger("\common\database\Connector");
+        $this->createConnection();
+    }
+
+    /**
+     * The only public method for consumers to either create or gain access to the sole singleton
+     * instance of the class.
+     *
+     * @return Connector instance
+     */
+    public static function Instance() {
+        static $singletonInstance = null;
+        if ($singletonInstance === null) {
+            $singletonInstance = new Connector();
```

Comment:
Hi Brady,

I'm not sure I follow you 100%, but let's keep the discussion going until we get there. The idea is that we don't want to keep creating new instances of the connection for all consumers but to reuse the same instance (hence the singleton pattern). However, since there is no framework or IoC container at the moment, this code doesn't provide too much value. I simply wrote it to be "future proof".

As for unit testing, the connector/doctrine itself would be mocked out.

Thanks,
Matthew